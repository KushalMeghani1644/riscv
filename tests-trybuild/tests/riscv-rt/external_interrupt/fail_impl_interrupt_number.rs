#[riscv_rt::external_interrupt(riscv::interrupt::Interrupt::SupervisorSoft)]
fn my_interrupt() {}

fn main() {}
