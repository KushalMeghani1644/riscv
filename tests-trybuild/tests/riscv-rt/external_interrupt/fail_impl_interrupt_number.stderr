error[E0277]: the trait bound `riscv::interrupt::Interrupt: ExternalInterruptNumber` is not satisfied
 --> tests/riscv-rt/external_interrupt/fail_impl_interrupt_number.rs:1:32
  |
1 | #[riscv_rt::external_interrupt(riscv::interrupt::Interrupt::SupervisorSoft)]
  | -------------------------------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^--
  | |                              |
  | |                              the trait `ExternalInterruptNumber` is not implemented for `riscv::interrupt::Interrupt`
  | required by a bound introduced by this call
  |
note: required by a bound in `assert_impl`
 --> tests/riscv-rt/external_interrupt/fail_impl_interrupt_number.rs:1:1
  |
1 | #[riscv_rt::external_interrupt(riscv::interrupt::Interrupt::SupervisorSoft)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `assert_impl`
  = note: this error originates in the attribute macro `riscv_rt::external_interrupt` (in Nightly builds, run with -Z macro-backtrace for more info)
